test/test.cpp

```diff
@@ -0,0 +1,120 @@
+#define CATCH_CONFIG_MAIN // This tells Catch to provide a main() - only do this in one cpp file
+#include "catch_amalgamated.hpp"
+#include "../Car.hpp"
+#include "../ElectricCar.hpp"
+#include "../ElectricEngine.hpp"
+#include "../HybridCar.hpp"
+#include "../PetrolCar.hpp"
+#include "../PetrolEngine.hpp"
+#include <iostream>
+
+SCENARIO("Testing Electric car")
+{
+    GIVEN("New electric car from Tesla")
+    {
+
+        ElectricCar tesla(std::make_unique<ElectricEngine>(130, 650));
+        Car *car = &tesla;
+
```

Comment:
You didn't test it well If you didn't catch a problem with reaching speed limit above 100 :) (check my issue in car.cpp)